src/Core/SonarArtefactKeys.cs

```diff
@@ -0,0 +1,73 @@
+﻿/*
```

Comment:
I renamed `SonarLanguageKeys` to `SonarArtefactKeys` and added the other constants and comments.

We have a couple of problems with constants:
* the same constants are defined in multiple places
* confusion between different types of artefact that happen to have the same values e.g. the C++ language and repo keys.

There's more to do, but this is a first step.